app/components/user-status-modal.hbs

```diff
@@ -2,31 +2,30 @@
 <div class="overlay">
   <div class="modal" data-test-modal>
     <div class="modal__inputs">
-      <label class="modal__inputs--description" data-test-label-from>From</label>
-      <Input name='fromDate' type='date' value={{this.fromDate}} {{on "change" this.updateValue}}
-        data-test-date-picker-from />
-      
       {{#if (eq @newStatus 'OOO')}}
+      
+      <label class="modal__inputs--description" data-test-label-from>From</label>
+      <Input name='fromDate' type='date' min={{this.disableDatesPrior}} value={{this.fromDate}} {{on "change"
+        this.updateValue}} data-test-date-picker-from />
       <label class="modal__inputs--description" data-test-label-until>Until</label>
-      <Input name='untilDate' type='date' value={{this.untilDate}} {{on "change" this.updateValue}}
-        data-test-date-picker-until />
-      {{/if}}
-
-      {{#if (not-eq @newStatus 'ACTIVE')}}
-      {{#if (eq @newStatus 'OOO')}}
+      <Input name='untilDate' type='date' min={{if this.fromDate this.fromDate this.disableDatesPrior}}
+        value={{this.untilDate}} {{on "change" this.updateValue}} data-test-date-picker-until />
       <label class="modal__inputs--description" data-test-label-reason>Reason</label>
-      {{/if}}
 
-      {{#if (eq @newStatus 'IDLE')}}
-      <label class="modal__inputs--description" data-test-label-missing-skills>What skills are you willing to learn</label>
+      {{else if (eq @newStatus 'IDLE')}}
```

Comment:
You can use the constants here for `IDLE` by defining the constant in the component file.